python/lsst/daf/persistence/access.py

```diff
@@ -0,0 +1,172 @@
+        path = os.path.join(self.root, 'repoCfg.yaml')
+        repoCfg.dumpToFile(path)
+        # with open(path, "wb") as outfile:
+        #     cPickle.dump(repoCfg, outfile, cPickle.HIGHEST_PROTOCOL)
+
+    def getCfg(self):
+        if not self.root:
+            raise RuntimeError("Storage root was declared to be None.")
+        path = os.path.join(self.root, 'repoCfg.yaml')
+        return Policy(filePath=path)
+
+    def write(self, butlerLocation, obj):
+        # self.log.log(pexLog.Log.DEBUG, "Put type=%s keys=%s to %s" % (datasetType, dataId, str(location)))
+        additionalData = butlerLocation.getAdditionalData()
+        storageName = butlerLocation.getStorageName()
+        locations = butlerLocation.getLocations()
+        with SafeFilename(locations[0]) as locationString:
+            logLoc = LogicalLocation(locationString, additionalData)
+
+            if storageName == "PickleStorage":
+                with open(logLoc.locString(), "wb") as outfile:
+                    cPickle.dump(obj, outfile, cPickle.HIGHEST_PROTOCOL)
+                return
+
+            if storageName == "ConfigStorage":
+                obj.save(logLoc.locString())
+                return
+
+            if storageName == "FitsCatalogStorage":
+                flags = additionalData.getInt("flags", 0)
+                obj.writeFits(logLoc.locString(), flags=flags)
+                return
+
+            # Create a list of Storages for the item.
+            storageList = StorageList()
+            storage = self.persistence.getPersistStorage(storageName, logLoc)
+            storageList.append(storage)
+
+            if storageName == 'FitsStorage':
+                self.persistence.persist(obj, storageList, additionalData)
+                return
+
+            # Persist the item.
+            if hasattr(obj, '__deref__'):
+                # We have a smart pointer, so dereference it.
+                self.persistence.persist(obj.__deref__(), storageList, additionalData)
+            else:
+                self.persistence.persist(obj, storageList, additionalData)
+
+class Access:
+    """Implements an butler framework interface for Transport, Storage, and Registry"""
+
+    @classmethod
+    def cfg(cls, storageCfg):
+        return Policy({'storageCfg': storageCfg})
+
+    def __init__(self, cfg):
+        self.storage = cfg['storageCfg.cls'](cfg['storageCfg'])
+
+    def mapperClass(self):
+        return self.storage.mapperClass()
+
+    def root(self):
+        """Root refers to the 'top' of a persisted repository.
+        Presumably its form can vary depending on the type of storage although currently the only Butler
+        storage that exists is posix, and root is the repo path."""
+        return self.storage.root
+
+    def setCfg(self, repoCfg):
+        self.storage.setCfg(repoCfg)
+
+    def getCfg(self):
+        return self.storage.getCfg()
+
+    def write(self, butlerLocation, obj):
```

Comment:
read is still implemented in Butler. Again, it wasn't required for this story (because ButlerLocation contains all the needed information). I was going to put off touching this until needed, e.g. DM-4542 "Refactor butler dispatch (object serialization) to be pluggable."